### DISCUSSION TITLE
How to disable go mod vendor?

### DISCUSSION THREAD

**Discussion by ngshiheng** [renovatebot/renovate#14610] [Request Help]
I tried searching through the discussions and [doc](https://docs.renovatebot.com/golang/) but I couldn't find anything relevant: 

Current Renovate behaviour:
- Flow: Fetch updates -> `go get` -> `go mod vendor` -> `postUpgradeTasks`

My use case:
- I want to run a custom script/cmd before I run `go mod vendor`
- What I have on my mind is that as long as I can disable `go mod vendor` here, the rest can be handled by `postUpgradeTasks`
- Flow: Fetch updates -> `go get` -> some custom script -> `postUpgradeTasks` (run `go mod vendor` here instead)

Questions:
- Is there a way I can change the order of how automated Dependency updates work for Go Modules?
- Or can I disable a certain step, for example step 9. (`go mod vendor`) as shown in the doc link/example above?

**Comment by rarkins** [comment]
Would a preUpgradeTask be a better solution for you if it existed? Ie then go mod vendor could run like usual

**Reply by ngshiheng** [reply]
Not exactly for my case because the custom script/cmd that I want to run has to happen after `go get` and right before `go mod vendor` :/

Though `preUpgradeTask` does sound like a pretty good feature; I'm sure someone will find it useful.
